cql3/expr/prepare_expr.cc

```diff
@@ -870,8 +870,8 @@ try_prepare_expression(const expression& expr, data_dictionary::database db, con
         [] (const constant&) -> std::optional<expression> {
             on_internal_error(expr_logger, "Can't prepare constant_value, it should not appear in parser output");
         },
-        [&] (const binary_operator&) -> std::optional<expression> {
-            on_internal_error(expr_logger, "binary_operators are not yet reachable via prepare_expression()");
+        [&] (const binary_operator& binop) -> std::optional<expression> {
+            return prepare_binary_operator(binop, db, *schema_opt);
```

Comment:
Missing a type check against `receiver`. If it's supplied, then it must be boolean_type.

This causes a type error if you later write `UPDATE foo SET int_col = ? > ?`.